### PR TITLE
Fix for use_case tests

### PR DESCRIPTION
**Pull request by CaaarolXu**
I did the final cleaning. Good luck with final exams!

-

There are 2 problems left for the tests:
1. execute test for GridInteractor
2. cover the catch part in "try...catch..." in DiscussionInteractor

-

New tests are added for test coverage. The tests with TODO comments can't be checked without the connection, so I don't know if they work properly for now.

-

Presently,
- Fix the problems that appeared in the tests. Now all tests should pass.
- Add one more test in LoginInteractorTest for coverage.
- Add one comment to SignupInteractorTest for how to use this test.

### CODE REVIEW

**Review by RaghavAtRuntime — DISMISSED** [review]
Fixes bugs in tests. Great!

**Comment by CaaarolXu** [comment]
Update for test coverage. Still waiting for the connection to check some tests.

**Review by majidism006 — DISMISSED** [review]
Perfect! Im glad the tests are complete. I can see that there is one more test for increased coverage and that all tests are now passing. Thanks Carol!

**Review by RaghavAtRuntime — DISMISSED** [review]

**Review by ibrmaj — DISMISSED** [review]

File: src/main/java/app/AppBuilder.java [hunk 1]
```diff
@@ -97,7 +99,7 @@ public class AppBuilder {
 
     public AppBuilder() {
         cardPanel.setLayout(cardLayout);
-        cardPanel.setPreferredSize(new Dimension(600, 500));
+        cardPanel.setPreferredSize(new Dimension(WIDTH, HEIGHT));
```
Comment (majidism006) [inline src/main/java/app/AppBuilder.java:102]: Thanks for creating the private static variables for the dimensions width and height, code looks alot cleaner now!

File: src/test/java/use_case/DiscussionInteractorTest.java [hunk 2]
```diff
@@ -60,84 +70,112 @@ public DiscussionPost addPostTemp(DiscussionPostInputData inputData) {
...
+        UserRepositoryImpl userRepository = new UserRepositoryImpl();
+
+        DiscussionPost post = FETCHInteractor.fetchRandomQuoteTemp(String.valueOf(
+                userRepository.findUserByUsername(USERNAME).getId()));
+        List<DiscussionPost> posts = repository.getAllPosts();
+        assertEquals(post.getUserId(), posts.get(posts.size() - 1).getUserId());
+        assertEquals(post.getContent(), posts.get(posts.size() - 1).getContent());
+    }
+
+    //TODO: check after connection works
```
Comment (majidism006) [inline src/test/java/use_case/DiscussionInteractorTest.java]: I believe the connection to the database has resumed. Is it possible to update this comment now?

File: src/test/java/use_case/DiscussionInteractorTest.java [hunk 3]
```diff
@@ -1,143 +1,112 @@
...
-                } catch (SQLException e) {
-                    e.printStackTrace();
-                    // Handle error
-                }
-                return null;
-            }
-        }
-
-        TempInteractor ADDInteractor = new TempInteractor(repository, OutputBoundary);
+        DiscussionPostViewModel viewModel = new DiscussionPostViewModel();
```
Comment (ibrmaj) [inline src/test/java/use_case/DiscussionInteractorTest.java:46]: When naming viewModel name it discussionViewModel for better readibilty

File: src/test/java/use_case/DiscussionInteractorTest.java [hunk 4]
```diff
@@ -1,143 +1,112 @@
...
-//        UserRepositoryImpl userRepository = new UserRepositoryImpl();
-//
-//        DiscussionPost post = FETCHInteractor.fetchRandomQuoteTemp(String.valueOf(
-//                userRepository.findUserByUsername(USERNAME).getId()));
-//        List<DiscussionPost> posts = repository.getAllPosts();
-//        assertEquals(post.getUserId(), posts.get(posts.size() - 1).getUserId());
-//        assertEquals(post.getContent(), posts.get(posts.size() - 1).getContent());
-//    }
+
+    @Test
```
Comment (ibrmaj) [inline src/test/java/use_case/DiscussionInteractorTest.java:65]: please add a small description of what each test is testing

File: src/test/java/use_case/HistoryInteractorTest.java [hunk 5]
```diff
@@ -17,21 +18,38 @@
 public class HistoryInteractorTest {
 
     public static final String USERNAME = "username";
-    public static final String TEXT = "Only for tesing";
+    public static final String TEXT = "Only for testing";
+    public static final String TEXT2 = "another text for testing";
```
Comment (ibrmaj) [inline src/test/java/use_case/HistoryInteractorTest.java:22]: Change variable name to be more specific

File: src/test/java/use_case/InstructionsInteractorTest.java [hunk 6]
```diff
@@ -83,8 +78,104 @@ void getRandomWordTest() {
...
+        InstructionsOutputBoundary instructionsOutputBoundary = new InstructionsPresenter(viewManagerModel,
+                instructionsViewModel, gridViewModel, discussionPostViewModel, profileViewModel);
+        InstructionsInputBoundary interactor = new InstructionsInteractor(userService,
+                instructionsOutputBoundary, wordRepository, gridState);
+
+        interactor.switchToDiscussionBoardView();
+
+        assertEquals("discussion",viewManagerModel.getState());
+    }
+
```
Comment (ibrmaj) [inline src/test/java/use_case/InstructionsInteractorTest.java:163]: small description for the test please

**Review by ibrmaj — CHANGES_REQUESTED** [review]

**Review by majidism006 — DISMISSED** [review]

**Review by ibrmaj — APPROVED** [review]

**Review by majidism006 — APPROVED** [review]

**Pull request merged**